<?xml version="1.0" ?>
<robot xmlns:xacro="http://www.ros.org/wiki/xacro" name="firefighter">
    <xacro:property name="width" value=".2" />
    <link name="base">
        <visual>
            <geometry>
                <mesh filename="package://mycobot_description/urdf/mercury_a1/base.dae" />
            </geometry>
            <origin xyz="0.0 0 0 " rpy=" 0 0 0" />
        </visual>
        <collision>
            <geometry>
                <mesh filename="package://mycobot_description/urdf/mercury_a1/base.dae" />
            </geometry>
            <origin xyz="0.0 0 0 " rpy=" 0 0 0" />
        </collision>
        
<inertial>
            <mass value="0.1" />
            <inertia ixx="0.03" iyy="0.03" izz="0.03" ixy="0.0" ixz="0.0" iyz="0.0" />
        </inertial>

    </link>
    <link name="joint1">
        <visual>
            <geometry>
                <mesh filename="package://mycobot_description/urdf/mercury_a1/joint1.dae" />
            </geometry>
            <origin xyz="0.0 0 -0.08 " rpy=" 0 0 0" />
        </visual>
        <collision>
            <geometry>
                <mesh filename="package://mycobot_description/urdf/mercury_a1/joint1.dae" />
            </geometry>
            <origin xyz="0.0 0 -0.08 " rpy=" 0 0 0" />
        </collision>
        
<inertial>
            <mass value="0.1" />
            <inertia ixx="0.03" iyy="0.03" izz="0.03" ixy="0.0" ixz="0.0" iyz="0.0" />
        </inertial>

    </link>
    <link name="joint2">
        <visual>
            <geometry>
                <mesh filename="package://mycobot_description/urdf/mercury_a1/joint2.dae" />
            </geometry>
            <origin xyz="0.0 0 0.0 " rpy=" 3.14159 0 -1.5708 " />
        </visual>
        <collision>
            <geometry>
                <mesh filename="package://mycobot_description/urdf/mercury_a1/joint2.dae" />
            </geometry>
            <origin xyz="0.0 0 0.0 " rpy=" 3.14159 0 -1.5708" />
        </collision>
        
<inertial>
            <mass value="0.1" />
            <inertia ixx="0.03" iyy="0.03" izz="0.03" ixy="0.0" ixz="0.0" iyz="0.0" />
        </inertial>

    </link>
    <link name="joint3">
        <visual>
            <geometry>
                <mesh filename="package://mycobot_description/urdf/mercury_a1/joint3.dae" />
            </geometry>
            <origin xyz="0.0 0 0.038 " rpy=" 0 0 0" />
        </visual>
        <collision>
            <geometry>
                <mesh filename="package://mycobot_description/urdf/mercury_a1/joint3.dae" />
            </geometry>
            <origin xyz="0.0 0 0.038" rpy=" 0 0 0" />
        </collision>
        
<inertial>
            <mass value="0.1" />
            <inertia ixx="0.03" iyy="0.03" izz="0.03" ixy="0.0" ixz="0.0" iyz="0.0" />
        </inertial>

    </link>
    <link name="joint4">
        <visual>
            <geometry>
                <mesh filename="package://mycobot_description/urdf/mercury_a1/joint4.dae" />
            </geometry>
            <origin xyz="0.0 0 0 " rpy=" 0 0 -1.5708" />
        </visual>
        <collision>
            <geometry>
                <mesh filename="package://mycobot_description/urdf/mercury_a1/joint4.dae" />
            </geometry>
            <origin xyz="0.0 0 0 " rpy=" 0 0 -1.5708" />
        </collision>
        
<inertial>
            <mass value="0.1" />
            <inertia ixx="0.03" iyy="0.03" izz="0.03" ixy="0.0" ixz="0.0" iyz="0.0" />
        </inertial>

    </link>
    <link name="joint5">
        <visual>
            <geometry>
                <mesh filename="package://mycobot_description/urdf/mercury_a1/joint5.dae" />
            </geometry>
            <origin xyz="0 0 -0.069 " rpy=" 0 0 0" />
        </visual>
        <collision>
            <geometry>
                <mesh filename="package://mycobot_description/urdf/mercury_a1/joint5.dae" />
            </geometry>
            <origin xyz="0 0 -0.069 " rpy=" 0 0 0" />
        </collision>
        
<inertial>
            <mass value="0.1" />
            <inertia ixx="0.03" iyy="0.03" izz="0.03" ixy="0.0" ixz="0.0" iyz="0.0" />
        </inertial>

    </link>
    <link name="joint6">
        <visual>
            <geometry>
                <mesh filename="package://mycobot_description/urdf/mercury_a1/joint6.dae" />
            </geometry>
            <origin xyz="0.0 0 -0.025 " rpy=" 0 0 -1.5708" />
        </visual>
        <collision>
            <geometry>
                <mesh filename="package://mycobot_description/urdf/mercury_a1/joint6.dae" />
            </geometry>
            <origin xyz="0.0 0 -0.025 " rpy=" 0 0 -1.5708" />
        </collision>
        
<inertial>
            <mass value="0.1" />
            <inertia ixx="0.03" iyy="0.03" izz="0.03" ixy="0.0" ixz="0.0" iyz="0.0" />
        </inertial>

    </link>
    <link name="joint7">
        <visual>
            <geometry>
                <mesh filename="package://mycobot_description/urdf/mercury_a1/joint7.dae" />
            </geometry>
            <origin xyz="0.0 0 -0.002 " rpy=" 0 0 -1.5708" />
        </visual>
        <collision>
            <geometry>
                <mesh filename="package://mycobot_description/urdf/mercury_a1/joint7.dae" />
            </geometry>
            <origin xyz="0.0 0 -0.002 " rpy=" 0 0 -1.5708" />
        </collision>
        
<inertial>
            <mass value="0.1" />
            <inertia ixx="0.03" iyy="0.03" izz="0.03" ixy="0.0" ixz="0.0" iyz="0.0" />
        </inertial>

    </link>
    <joint name="joint1_to_base" type="revolute">
        <axis xyz="0 0 1" />
        <limit effort="1000.0" lower="-2.8797" upper="2.8797" velocity="0" />
        <parent link="base" />
        <child link="joint1" />
        <origin xyz="0 0 0.175" rpy="0 0 0" />
    </joint>
    <joint name="joint2_to_joint1" type="revolute">
        <axis xyz="0 0 1" />
        <limit effort="1000.0" lower="-1.7453" upper="1.7453" velocity="0" />
        <parent link="joint1" />
        <child link="joint2" />
        <!-- <origin xyz= "0 0  0" rpy = "1.5708 3.14159 0"/>   -->
        <origin xyz="0 0  0" rpy="-1.5708 0 0" />
    </joint>
    <joint name="joint3_to_joint2" type="revolute">
        <axis xyz=" 0 0 1" />
        <limit effort="1000.0" lower="-2.8797" upper="2.8797" velocity="0" />
        <parent link="joint2" />
        <child link="joint3" />
        <origin xyz="0 -0.11 0   " rpy="1.5708 0 0" />
    </joint>
    <joint name="joint4_to_joint3" type="revolute">
        <axis xyz=" 0 0 1" />
        <limit effort="1000.0" lower="-3.0543" upper="0.0174" velocity="0" />
        <parent link="joint3" />
        <child link="joint4" />
        <origin xyz="0.03 0 0.095" rpy="-1.5708 0 3.14159" />
    </joint>
    <joint name="joint5_to_joint4" type="revolute">
        <axis xyz="0 0 1" />
        <limit effort="1000.0" lower="-2.8797" upper="2.8797" velocity="0" />
        <parent link="joint4" />
        <child link="joint5" />
        <origin xyz="0.03 -0.126 0" rpy="1.5708 3.14159 0" />
    </joint>
    <joint name="joint6_to_joint5" type="revolute">
        <axis xyz="0 0 1" />
        <limit effort="1000.0" lower="-0.0174" upper="3.14159" velocity="0" />
        <parent link="joint5" />
        <child link="joint6" />
        <origin xyz="0 0.0 0.08" rpy="1.5708 0 0" />
    </joint>
    <joint name="joint7_to_joint6" type="revolute">
        <axis xyz="0 0 1" />
        <limit effort="1000.0" lower="-2.8797" upper="2.8797" velocity="0" />
        <parent link="joint6" />
        <child link="joint7" />
        <origin xyz="0.038 -0.046 0" rpy="1.5708 0 0" />
    </joint>
    
<transmission name="trans_joint1_to_base">
        <type>transmission_interface/SimpleTransmission</type>
        <joint name="joint1_to_base">
            <hardwareInterface>hardware_interface/EffortJointInterface</hardwareInterface>
        </joint>
        <actuator name="joint1_to_base_motor">
            <hardwareInterface>hardware_interface/EffortJointInterface</hardwareInterface>
            <mechanicalReduction>1</mechanicalReduction>
        </actuator>
    </transmission>
    <transmission name="trans_joint2_to_joint1">
        <type>transmission_interface/SimpleTransmission</type>
        <joint name="joint2_to_joint1">
            <hardwareInterface>hardware_interface/EffortJointInterface</hardwareInterface>
        </joint>
        <actuator name="joint2_to_joint1_motor">
            <hardwareInterface>hardware_interface/EffortJointInterface</hardwareInterface>
            <mechanicalReduction>1</mechanicalReduction>
        </actuator>
    </transmission>
    <transmission name="trans_joint3_to_joint2">
        <type>transmission_interface/SimpleTransmission</type>
        <joint name="joint3_to_joint2">
            <hardwareInterface>hardware_interface/EffortJointInterface</hardwareInterface>
        </joint>
        <actuator name="joint3_to_joint2_motor">
            <hardwareInterface>hardware_interface/EffortJointInterface</hardwareInterface>
            <mechanicalReduction>1</mechanicalReduction>
        </actuator>
    </transmission>
    <transmission name="trans_joint4_to_joint3">
        <type>transmission_interface/SimpleTransmission</type>
        <joint name="joint4_to_joint3">
            <hardwareInterface>hardware_interface/EffortJointInterface</hardwareInterface>
        </joint>
        <actuator name="joint4_to_joint3_motor">
            <hardwareInterface>hardware_interface/EffortJointInterface</hardwareInterface>
            <mechanicalReduction>1</mechanicalReduction>
        </actuator>
    </transmission>
    <transmission name="trans_joint5_to_joint4">
        <type>transmission_interface/SimpleTransmission</type>
        <joint name="joint5_to_joint4">
            <hardwareInterface>hardware_interface/EffortJointInterface</hardwareInterface>
        </joint>
        <actuator name="joint5_to_joint4_motor">
            <hardwareInterface>hardware_interface/EffortJointInterface</hardwareInterface>
            <mechanicalReduction>1</mechanicalReduction>
        </actuator>
    </transmission>
    <transmission name="trans_joint6_to_joint5">
        <type>transmission_interface/SimpleTransmission</type>
        <joint name="joint6_to_joint5">
            <hardwareInterface>hardware_interface/EffortJointInterface</hardwareInterface>
        </joint>
        <actuator name="joint6_to_joint5_motor">
            <hardwareInterface>hardware_interface/EffortJointInterface</hardwareInterface>
            <mechanicalReduction>1</mechanicalReduction>
        </actuator>
    </transmission>
    <transmission name="trans_joint7_to_joint6">
        <type>transmission_interface/SimpleTransmission</type>
        <joint name="joint7_to_joint6">
            <hardwareInterface>hardware_interface/EffortJointInterface</hardwareInterface>
        </joint>
        <actuator name="joint7_to_joint6_motor">
            <hardwareInterface>hardware_interface/EffortJointInterface</hardwareInterface>
            <mechanicalReduction>1</mechanicalReduction>
        </actuator>
    </transmission>
    <gazebo>
        <plugin name="gazebo_ros_control" filename="libgazebo_ros_control.so">
            <robotNamespace>/</robotNamespace>
        </plugin>
    </gazebo>
</robot>